### PR TITLE
[s] Caps misfire probability to prevent infinite recursive procs

### PR DESCRIPTION
**Pull request by necromanceranne**
## About The Pull Request

Not the proper fix for this but it at least prevents the exploitative behaviour. 

## Why It's Good For The Game

I did a nyoops.

## Changelog
:cl:
fix: Prevents guns that can misfire reaching 100% probability.
/:cl:

### CODE REVIEW

File: code/modules/projectiles/guns/ballistic.dm [hunk 1]
```diff
@@ -104,12 +104,17 @@
...
+	/// Misfire Variables ///
+
 	/// Whether our ammo misfires now or when it's set by the wrench_act. TRUE means it misfires.
 	var/can_misfire = FALSE
 	///How likely is our gun to misfire?
 	var/misfire_probability = 0
 	///How much does shooting the gun increment the misfire probability?
 	var/misfire_percentage_increment = 0
+	///What is the cap on our misfire probability? Do not set this to 100.
+	var/misfire_probability_cap = 25
```
Comment (ATH1909) [inline code/modules/projectiles/guns/ballistic.dm:117]: detective gaming

**Comment by ATH1909** [comment]
fr though increase the cap here to like 50 or 75 or something or else dets are gonna abuse the hell out of this

**Comment by necromanceranne** [comment]
25% is still 1 in 4 shots and they can already fire one bullet without issue before cleaning. Keeping this relatively low also helps prevent too many loops until I have time to fix it proper.

**Comment by LemonInTheDark** [comment]
We'll do this for now, and get something more stable later

**Pull request merged**